### ISSUE TITLE
Can we have option to proxy similar to nginx

### ISSUE THREAD

**Issue by iknityanand** [vercel/serve#602]
Hi team,

Is there an option or documentation on how to proxy request using serve.
I am looking for reverse proxy and load balancing options here.

**Comment by Morriz** [comment]
yeah, I just like to test builds on my laptop and need to proxypass to an api....very valid use case

**Comment by heidemn** [comment]
You could consider as a temp solution to use e.g. Caddy as entry point / reverse proxy:
https://caddyserver.com/docs/caddyfile/patterns#reverse-proxy

```
                      /--> /app/... (npx serve)
request -> Caddy proxy
                      \--> /api1/... (e.g. Nginx)
                      \--> /api2/... (e.g. Apache)
```

**Comment by warren-bank** [comment]
_shameless self-promotion alert_:<br>this feature is available in my [`@warren-bank/serve`](https://github.com/warren-bank/node-serve) fork of `serve`

notes about using [redirect](https://github.com/warren-bank/node-serve/blob/master/lib/serve-handler/README.md#redirects-array) rules:
* require an `engine` attribute..<br>which permits you to choose which methodology/library is used to:
  - match the url pathname against a pattern
  -  replace capture groups
* an optional boolean `proxy` attribute can be used to:
  - download from an external url
  - pipe the response back to the client

**Comment by chenweiyi** [comment]
可以使用vite做静态服务，并且支持proxy , 只需配置vite.config.ts:
```
import { defineConfig } from 'vite';
export default defineConfig(({ mode }) => {
  return {
    preview: {
      proxy: {
        '^/xxx': {
          target: 'https://xxx',
          changeOrigin: true,
          ws: false,
          rewrite: path => path.replace(/^\/xxx/, '/xxx'),
        },
      },
    },
  };
});
```
